[BUG] E-Mail workflow attachment not found

### Description

When in a workflow attached pdf is activated, i got a file not found error in the logfile. 

[2025-01-14 10:03:56,134] [ERROR] [paperless.handlers] Error occurred sending notification email: [Errno 2] No such file or directory: '/usr/src/paperless/media/documents/originals/0000565.pdf'

### Steps to reproduce

1. add a workflow with attached pdf activated and trigger created/new document

### Webserver logs

```bash
[2025-01-14 10:03:56,132] [INFO] [paperless.handlers] Applying WorkflowAction 2 from Workflow: TEST1

[2025-01-14 10:03:56,134] [ERROR] [paperless.handlers] Error occurred sending notification email: [Errno 2] No such file or directory: '/usr/src/paperless/media/documents/originals/0000565.pdf'

Traceback (most recent call last):

  File "/usr/src/paperless/src/documents/signals/handlers.py", line 946, in email_action

    email.attach_file(document.source_path)

  File "/usr/local/lib/python3.12/site-packages/django/core/mail/message.py", line 355, in attach_file

    with path.open("rb") as file:

         ^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/pathlib.py", line 1013, in open

    return io.open(self, mode, buffering, encoding, errors, newline)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

FileNotFoundError: [Errno 2] No such file or directory: '/usr/src/paperless/media/documents/originals/0000565.pdf'
```


### Browser logs

_No response_

### Paperless-ngx version

2.14.1

### Host OS

Ubuntu

### Installation method

Docker - official image

### System status

_No response_

### Browser

_No response_

### Configuration changes

_No response_

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

--- a/src/documents/signals/handlers.py
+++ b/src/documents/signals/handlers.py
@@ -943,7 +943,10 @@ def run_workflows(
                 to=action.email.to.split(","),
             )
             if action.email.include_document:
-                email.attach_file(document.source_path)
+                # Something could be renaming the file concurrently so it can't be attached
+                with FileLock(settings.MEDIA_LOCK):
+                    document.refresh_from_db()
+                    email.attach_file(document.source_path)
             n_messages = email.send()
             logger.debug(
                 f"Sent {n_messages} notification email(s) to {action.email.to}",

